languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -424,6 +424,136 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
 (the rule on "ajouter en plus"
 was already there before the category was created,
 though, and has since been slightly modified)-->
+      <rulegroup id="PLEONASMES_DIVERS" name="pléonasmes divers" default="temp_off">
+         <rule>
+          <pattern>
+              <token>autrement</token>
+              <token>plus</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
+          <suggestion>\1</suggestion>
+          <url>https://www.projet-voltaire.fr/regles-orthographe/autrement-ou-autrement-plus/</url>
+          <example correction="autrement">Il est <marker>autrement plus</marker> lent.</example>
+      </rule>
+      <rule>
+          <pattern>
+              <token>bip</token>
+              <token>sonore</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
+          <suggestion>\1</suggestion>
+          <suggestion>signal sonore</suggestion>
+          <example correction="bip|signal sonore">Ton téléphone a émis un <marker>bip sonore</marker>.</example>
+      </rule>
+      <rule>
+          <pattern>
+              <token>krach</token>
+              <token>boursier</token>
+          </pattern>
+          <message>Cette expression peut être considérée comme un pléonasme.</message>
+          <suggestion>\1</suggestion>
+          <example correction="krach">En 1929, il y a eu un terrible <marker>krach boursier</marker>.</example>
+      </rule>
+      <rule>
+          <pattern>
+              <token>légère</token>
+              <token>collation</token>
+          </pattern>
+          <message>Cette expression est un pléonasme.</message>
```

Comment:
Careful with any pleonasm rules. If two words occur frequently next to each other (collocations / expressions figées), we must allow them even if their meanings are somewhat tautological. This is especially true if they occur frequently in formal written speech.

All the messages have to be a lot softer. "Cette expression est considérée pléonastique, car une collation est toujours légère." (or something like that).  This rule about "little snacks" shows you why that is: If the speaker decides to say that they had not a regular "snack" but a "little snack", they might want to say that their snack was particularly tiny.

Pleonasm rules tend to be to prescriptive (and are usually not appreciated by users, just like anglicism rules). We should only write them for blatant, uncommon "abus de langage".

Reply:
I modified the message so it is more friendly and explains the rule.